controllers/project_mesh_controller_test.go

```diff
@@ -228,23 +224,80 @@ var _ = When("Namespace is created", Label(labels.EvnTest), func() {
 				},
 			}
 
-			defer objectCleaner.DeleteAll(route, istioNs)
-
-			// when
 			Expect(cli.Create(context.Background(), istioNs)).To(Succeed())
 			Expect(cli.Create(context.Background(), route)).To(Succeed())
+		})
+
+		AfterEach(func() {
+			objectCleaner.DeleteAll(route, istioNs)
+		})
+
+		It("should add just gateway name to the namespace if there is no gateway namespace defined", func() {
+			// given
+			testNs = &v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "plain-meshified-ns",
+					Annotations: map[string]string{
+						controllers.AnnotationServiceMesh: "true",
+					},
+				},
+			}
+
+			// update route to remove annotations
+			route.Labels["maistra.io/gateway-namespace"] = ""
+			Expect(cli.Update(context.Background(), route)).To(Succeed())
+
+			// when
+			Expect(cli.Create(context.Background(), testNs)).To(Succeed())
 
 			// then
-			actualRoute := &routev1.Route{}
-			Eventually(func() error {
-				return cli.Get(context.Background(), types.NamespacedName{
-					Namespace: route.Namespace,
-					Name:      route.Name,
-				}, actualRoute)
+			actualTestNs := &v1.Namespace{}
+			Eventually(func() string {
+				_ = cli.Get(context.Background(), types.NamespacedName{Name: testNs.Name}, actualTestNs)
+				return actualTestNs.Annotations[controllers.AnnotationGateway]
 			}).
 				WithTimeout(timeout).
 				WithPolling(interval).
-				Should(Succeed())
```

Comment:
Removed from all the tests, as this is not needed. We create route in `BeforeEach`